packages/cli/src/cli/utils/delta.ts

```diff
@@ -0,0 +1,100 @@
+import _ from "lodash";
+import z from "zod";
+import { MD5 } from "object-hash";
+import { tryReadFile, writeFile, checkIfFileExists } from "../utils/fs";
+import * as path from "path";
+import YAML from "yaml";
+
+const LockSchema = z.object({
+  version: z.literal(1).default(1),
+  checksums: z
+    .record(
+      z.string(), // localizable files' keys
+      // checksums hashmap
+      z
+        .record(
+          // key
+          z.string(),
+          // checksum of the key's value in the source locale
+          z.string(),
+        )
+        .default({}),
+    )
+    .default({}),
+});
+export type LockData = z.infer<typeof LockSchema>;
+
+export function createDeltaProcessor(fileKey: string) {
+  const lockfilePath = path.join(process.cwd(), "i18n.lock");
+  return {
+    async checkIfLockExists() {
+      return checkIfFileExists(lockfilePath);
+    },
+    async calculateDelta(params: {
+      sourceData: Record<string, any>;
+      targetData: Record<string, any>;
+      checksums: Record<string, string>;
+    }) {
+      let added = _.difference(Object.keys(params.sourceData), Object.keys(params.targetData));
+      let removed = _.difference(Object.keys(params.targetData), Object.keys(params.sourceData));
+      const updated = _.filter(Object.keys(params.sourceData), (key) => {
+        return MD5(params.sourceData[key]) !== params.checksums[key] && params.checksums[key];
+      });
+
+      // Find renamed keys - keys that exist in both but with different names (same content)
```

Comment:
Keys do not exist in both, value hashes do, right?

```suggestion
      // Find renamed keys - hashes that exist in both (same content) but with different keys
```